src/argilla/server/search_engine/base.py

```diff
@@ -96,3 +96,24 @@ async def search(
         limit: int = 100,
     ) -> SearchResponses:
         pass
+
+    @abstractmethod
+    async def configure_index_vectors(self, vector_settings: VectorSettings):
+        pass
+
+    @abstractmethod
+    async def set_records_vectors(self, dataset: Dataset, vectors: Iterable[Vector]):
+        pass
+
+    @abstractmethod
+    async def similarity_search(
+        self,
+        dataset: Dataset,
+        vector_settings: VectorSettings,
+        value: Optional[List[float]] = None,
+        record: Optional[Record] = None,
+        user_response_status_filter: Optional[UserResponseStatusFilter] = None,
+        max_results: conint(ge=2, le=500) = 100,
```

Comment:
No validation will be applied here by `pydantic`, so let's just use `int`

```suggestion
        max_results: int = 100,
```